tests/oauth2/test_service_account.py

```diff
@@ -481,6 +486,173 @@ def test_refresh_success(self, jwt_grant):
         # expired)
         assert credentials.valid
 
+    @mock.patch("google.oauth2._client.jwt_grant", autospec=True)
+    def test_refresh_success_with_trust_boundary(self, jwt_grant):
+        credentials = self.make_credentials()
+        credentials._enable_trust_boundary()
+        token = "token"
+        jwt_grant.return_value = (
+            token,
+            _helpers.utcnow() + datetime.timedelta(seconds=500),
+            {},
+        )
+        request = mock.create_autospec(transport.Request, instance=True)
+
+        # Refresh credentials
```

Comment:
nit: Comments should describe why instead of what, so this comment is redundant.

[go/comments-codelab](https://goto.google.com/comments-codelab)